src/history/HistoryManagerImpl.cpp

```diff
@@ -251,7 +251,7 @@ void
 HistoryManagerImpl::queueCurrentHistory()
 {
     auto has = getLastClosedHistoryArchiveState();
-
+    has.resolveAnyReadyFutures();
```

Comment:
Should this be a bit tighter, i.e, in `getLastClosedHistoryArchiveState `?  (generally, I'm wondering what's the best place to do resolution to avoid having this issue elsewhere, e.g. by guaranteeing that latest HAS's state is up-to-date)

Reply:
@marta-lokhova it could definitely be moved inwards a bit in the logic; I avoided doing so out of expedience as many other code paths and tests are coupled to the choice of when to resolve and I was trying to get this done by the 11.2 release. As it happens we didn't take this for 11.2 so there's probably time to go back and fix all the things that are sensitive to the matter.

Reply:
@marta-lokhova ok, fixed! See if you like this variant better :)